### PR TITLE
Merge release v1.7.0 into develop

### PR DESCRIPTION
**Pull request by boonedev**
This PR:
- Resolves merge conficts
- Fixes namespace issues and adds use statements
- Adds REST_Api->init_hooks
- js linting

### CODE REVIEW

File: accessibility-checker.php [hunk 1]
```diff
@@ -97,12 +100,11 @@
 	include_once plugin_dir_path( __FILE__ ) . 'vendor/autoload.php';
 }
 
-use EDAC\Inc\Plugin;
```
Comment (boonedev) [inline accessibility-checker.php:100]: Moved to top of file.

File: accessibility-checker.php [hunk 2]
```diff
@@ -252,6 +254,11 @@ function edac_before_page_render() {
 
 		// Check the page if it hasn't already been checked.
 		global $post;
+		$post_id = is_object( $post ) ? $post->ID : null;
```
Comment (boonedev) [inline accessibility-checker.php:257]: broken rss fix.

File: admin/class-admin.php [hunk 3]
```diff
@@ -7,8 +7,6 @@
 
 namespace EDAC\Admin;
 
-use EDAC;
```
Comment (boonedev) [inline admin/class-admin.php:10]: Not needed.

File: admin/class-admin.php [hunk 4]
```diff
@@ -18,23 +16,22 @@ class Admin {
 	 * Class constructor.
 	 */
 	public function __construct() {
-		$this->init();
```
Comment (boonedev) [inline admin/class-admin.php:21]: Moved these to public functions to stay in keeping with only using constructor for variable inits.

File: admin/class-admin.php [hunk 5]
```diff
@@ -18,23 +16,22 @@ class Admin {
...
+	public function init() {
+		
+		$admin_notices = new Admin_Notices();
 		$admin_notices->init_hooks();
 
-		$widgets = new \EDAC\Admin\Widgets();
+		$widgets = new Widgets();
 		$widgets->init_hooks();
 
-		new EDAC\REST_Api();
```
Comment (boonedev) [inline admin/class-admin.php:37]: API needs to be reachable outside of Admin, so moved REST_Api init to class-plugin.php.

File: admin/class-admin.php [hunk 6]
```diff
@@ -47,10 +44,10 @@ private function init_ajax() {
 			return;
 		}
 
-		$ajax = new \EDAC\Admin\Ajax();
```
Comment (boonedev) [inline admin/class-admin.php:50]: Not needed.

File: admin/class-admin.php [hunk 7]
```diff
@@ -47,10 +44,10 @@ private function init_ajax() {
...
+		$ajax = new Ajax();
 		$ajax->init_hooks();
 
-		$frontend_highlight = new \EDAC\Admin\Frontend_Highlight();
```
Comment (boonedev) [inline admin/class-admin.php:53]: Not needed.

File: admin/class-issues-query.php [hunk 8]
```diff
@@ -5,10 +5,7 @@
  * @package Accessibility_Checker
  */
 
-namespace EDAC;
```
Comment (boonedev) [inline admin/class-issues-query.php:8]: Not needed.

File: admin/class-scans-stats.php [hunk 9]
```diff
@@ -5,12 +5,7 @@
  * @package Accessibility_Checker
  */
 
-namespace EDAC;
```
Comment (boonedev) [inline admin/class-scans-stats.php:8]: Not needed.

File: admin/class-scans-stats.php [hunk 10]
```diff
@@ -5,12 +5,7 @@
...
-
```
Comment (boonedev) [inline admin/class-scans-stats.php:9]: Not needed.

File: admin/class-scans-stats.php [hunk 11]
```diff
@@ -5,12 +5,7 @@
...
-use EDAC\Admin\Settings;
-use EDAC\Admin\Helpers;
-use EDAC\Issues_Query;
```
Comment (boonedev) [inline admin/class-scans-stats.php:12]: Not needed.

File: admin/class-issues-query.php [hunk 12]
```diff
@@ -5,10 +5,7 @@
...
-
-use EDAC\Admin\Helpers;
-use EDAC\Admin\Settings;
```
Comment (boonedev) [inline admin/class-issues-query.php:11]: Not needed.

File: admin/class-admin.php (hunk 7 above)
Comment (boonedev) [inline admin/class-admin.php:53]: Not needed.

File: admin/class-admin.php [hunk 13]
```diff
@@ -18,23 +16,22 @@ class Admin {
...
 	 * @return void
 	 */
-	private function init() {
-		$admin_notices = new \EDAC\Admin\Admin_Notices();
+	public function init() {
+		
+		$admin_notices = new Admin_Notices();
 		$admin_notices->init_hooks();
 
-		$widgets = new \EDAC\Admin\Widgets();
```
Comment (boonedev) [inline admin/class-admin.php:34]: Not needed.

File: accessibility-checker.php [hunk 14]
```diff
@@ -252,6 +254,11 @@ function edac_before_page_render() {
 
 		// Check the page if it hasn't already been checked.
 		global $post;
+		$post_id = is_object( $post ) ? $post->ID : null;       
+		if ( null === $post_id ) {
+			return;
+		}
```
Comment (boonedev) [inline accessibility-checker.php:260]: Broken rss fix.

File: admin/class-admin.php [hunk 15]
```diff
@@ -18,23 +16,22 @@ class Admin {
 	 * Class constructor.
 	 */
 	public function __construct() {
-		$this->init();
-		$this->init_ajax();
```
Comment (boonedev) [inline admin/class-admin.php:22]: Moved these to public functions to stay in keeping with only using constructor for variable inits.

File: admin/class-admin.php (hunk 5 above)
Comment (boonedev) [inline admin/class-admin.php:37]: API needs to be reachable outside of Admin, so moved REST_Api init to class-plugin.php.

File: admin/class-scans-stats.php [hunk 16]
```diff
@@ -156,7 +151,7 @@ public function summary() {
 
 		$data['public_post_types_count'] = (int) count( $post_types );
 
-		$issues_query = new \EDAC\Issues_Query( array(), $this->record_limit, Issues_Query::FLAG_INCLUDE_ALL_POST_TYPES );
```
Comment (boonedev) [inline admin/class-scans-stats.php:159]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 17]
```diff
@@ -171,7 +166,7 @@ function ( $item ) {
 		);
 
 		foreach ( $rule_slugs as $rule_slug ) {
-			$rule_query = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:174]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 18]
```diff
@@ -190,7 +185,7 @@ function ( $item ) {
 			$data['passed_percentage'] = round( ( $data['rules_passed'] / $this->rule_count ) * 100, 2 );
 		}
 
-		$warning_issues_query      = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:193]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 19]
```diff
@@ -200,7 +195,7 @@ function ( $item ) {
 		$data['warnings']          = (int) $warning_issues_query->count();
 		$data['distinct_warnings'] = (int) $warning_issues_query->distinct_count();
 
-		$contrast_issues_query = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:203]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 20]
```diff
@@ -211,7 +206,7 @@ function ( $item ) {
 		$data['contrast_errors']          = (int) $contrast_issues_query->count();
 		$data['distinct_contrast_errors'] = (int) $contrast_issues_query->distinct_count();
 
-		$error_issues_query      = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:214]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 21]
```diff
@@ -362,7 +357,7 @@ public function issues_summary_by_post_type( $post_type ) {
 
 		$data = array();
 
-		$error_issues_query      = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:365]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 22]
```diff
@@ -372,7 +367,7 @@ public function issues_summary_by_post_type( $post_type ) {
 		$data['errors']          = $error_issues_query->count();
 		$data['distinct_errors'] = $error_issues_query->distinct_count();
 
-		$warning_issues_query      = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:375]: Full NS not needed.

File: admin/class-scans-stats.php [hunk 23]
```diff
@@ -382,7 +377,7 @@ public function issues_summary_by_post_type( $post_type ) {
 		$data['warnings']          = $warning_issues_query->count();
 		$data['distinct_warnings'] = $warning_issues_query->distinct_count();
 
-		$color_contrast_issues_query      = new \EDAC\Issues_Query(
```
Comment (boonedev) [inline admin/class-scans-stats.php:385]: Full NS not needed.

File: admin/class-welcome-page.php [hunk 24]
```diff
@@ -9,8 +9,6 @@
 
 namespace EDAC\Admin;
 
-use EDAC\Scans_Stats;
```
Comment (boonedev) [inline admin/class-welcome-page.php:12]: Not needed.

File: includes/classes/class-plugin.php [hunk 25]
```diff
@@ -31,13 +36,14 @@ public function __construct() {
 	 * @return void
 	 */
 	private function init() {
-		$accessibility_statement = new \EDAC\Inc\Accessibility_Statement();
```
Comment (boonedev) [inline includes/classes/class-plugin.php:34]: Ful NS not needed.

File: includes/classes/class-plugin.php [hunk 26]
```diff
@@ -31,13 +36,14 @@ public function __construct() {
...
+
+		$accessibility_statement = new Accessibility_Statement();
 		$accessibility_statement->init_hooks();
 
-		$simplified_summary = new \EDAC\Inc\Simplified_Summary();
```
Comment (boonedev) [inline includes/classes/class-plugin.php:37]: Full NS not needed.

File: includes/classes/class-plugin.php [hunk 27]
```diff
@@ -31,13 +36,14 @@ public function __construct() {
...
+		$simplified_summary = new Simplified_Summary();
 		$simplified_summary->init_hooks();
 
-		$lazyload_filter = new \EDAC\Inc\Lazyload_Filter();
```
Comment (boonedev) [inline includes/classes/class-plugin.php:40]: Full NS not needed.

File: includes/classes/class-rest-api.php [hunk 28]
```diff
@@ -5,40 +5,41 @@
...
  * Class that initializes and handles the REST api
  */
 class REST_Api {
 
-	/**
-	 * If class has already been initialized.
-	 *
-	 * @var boolean
-	 */
-	private static $initialized = false;
```
Comment (boonedev) [inline includes/classes/class-rest-api.php:25]: Singleton not needed.

File: includes/classes/class-rest-api.php [hunk 29]
```diff
@@ -5,40 +5,41 @@
...
 
 	/**
 	 * Constructor
 	 */
 	public function __construct() {
+	}
 
-		if ( ! self::$initialized ) {
```
Comment (boonedev) [inline includes/classes/class-rest-api.php:32]: Not needed.

File: includes/classes/class-rest-api.php [hunk 30]
```diff
@@ -5,40 +5,41 @@
...
-			$this->initialize();
-		}
```
Comment (boonedev) [inline includes/classes/class-rest-api.php:34]: Singleton not needed.

File: includes/classes/class-rest-api.php [hunk 31]
```diff
@@ -124,23 +125,6 @@ function () use ( $ns, $version ) {
...
-						'methods'             => 'GET',
-						'callback'            => array( $this, 'get_scans_stats' ),
-						'permission_callback' => function () {
-							return current_user_can( 'read' ); // able to access the admin dashboard.
-						},
-					)
-				);
-			}
-		);
-
```
Comment (boonedev) [inline includes/classes/class-rest-api.php:143]: Bugfix - not needed. Closes: https://github.com/equalizedigital/accessibility-checker/pull/365

File: includes/classes/class-rest-api.php [hunk 32]
```diff
@@ -201,8 +185,7 @@ public function set_post_scan_results( $request ) {
 
 		$post_type  = get_post_type( $post );
 		$post_types = Helpers::get_option_as_array( 'edac_post_types' );
-
-		if ( ! empty( $post_types ) || ! in_array( $post_type, $post_types, true ) ) {
+		if ( empty( $post_types ) || ! in_array( $post_type, $post_types, true ) ) {
```
Comment (boonedev) [inline includes/classes/class-rest-api.php:188]: Bugfix: typo.

File: includes/classes/class-rest-api.php [hunk 33]
```diff
@@ -262,12 +245,12 @@ public function set_post_scan_results( $request ) {
 
 			do_action( 'edac_after_validate', $post_id, 'js' );
 
-			// Update the summary info that is stored in meta this post.
-			edac_summary( $post_id );
-
```
Comment (boonedev) [inline includes/classes/class-rest-api.php:267]: Bugfix. Moved to after edac_remove_corrected_posts is called.

File: includes/options-page.php [hunk 34]
```diff
@@ -5,8 +5,6 @@
  * @package Accessibility_Checker
  */
 
-use EDAC\Scans_Stats;
```
Comment (boonedev) [inline includes/options-page.php:8]: Not needed. Also, I think calling this outside a class could cause global NS conflicts.

File: includes/validate.php [hunk 35]
```diff
@@ -265,7 +271,7 @@ function edac_get_content( $post ) {
 		$parsed_url = wp_parse_url( home_url() );
 
 		if ( isset( $parsed_url['host'] ) ) {
-			$is_local_loopback = \EDAC\Admin\Helpers::is_domain_loopback( $parsed_url['host'] );
```
Comment (boonedev) [inline includes/validate.php:268]: Full NS not needed.

**Comment by SteveJonesDev** [comment]
@boonedev, the next and previous buttons on the front-end highlighting aren't working correctly.

**Review by SteveJonesDev — APPROVED** [review]
I tested frontend next and previous buttons and they are working now. Good to merge.

**Comment by boonedev** [comment]
@SteveJonesDev Updated with a fix.

**Pull request merged**